### PR TITLE
Correct independent scale scope for concat/repeat

### PR DESCRIPTION
**Pull request by kanitw**
Do not nest them so that signals can still refer to these scales.

Part of #2555

### CODE REVIEW

**Comment by kanitw** [comment]
This now works for [this concat spec](https://github.com/vega/vega-lite/issues/2555#issue-238020243). 

vega_editor

**Comment by kanitw** [comment]
Ready for review! --- Looking at each commit one-by-one would be easier :)

File: src/compile/model.ts [hunk 1]
```diff
@@ -241,7 +241,7 @@ export abstract class Model {
   public abstract assembleLayoutSignals(): VgSignal[];
 
   public assembleScales(): VgScale[] {
-    return assembleScale(this);
+    return assembleScalesForModel(this);
   }
```
Comment (arvind) [inline src/compile/model.ts]: As far as I can tell, `Model` and every `Model` subclass has an identical `assembleScales` function? My vote would be to remove them from the subclasses to reduce the number of codepaths.

File: src/compile/model.ts [hunk 2]
```diff
@@ -280,10 +280,10 @@ export abstract class Model {
     return assembleLegends(this);
   }
 
-  public assembleGroup(signals: VgSignal[] = []) {
+  public assembleGroup(mixins: {signals?: VgSignal[], scales: VgScale[]}) {
```
Comment (arvind) [inline src/compile/model.ts]: Perhaps worth adding a comment about why scales and signals are passed in (rather than assembled within this method as before)? Took me a second to understand why the change in logic was necessary (to allow for scale references from signals, which the previous approach did not).

File: src/compile/model.ts (hunk 1 above)
Comment (kanitw) [inline src/compile/model.ts]: They are actually different.  I'm making it `abstract` then so it's clearer.

File: src/compile/model.ts (hunk 2 above)
Comment (kanitw) [inline src/compile/model.ts]: I just found a way to make this part less convoluted! 
https://github.com/vega/vega-lite/pull/2698/commits/fbfc544565b0a248704bac95970d4734d7ada971

**Comment by kanitw** [comment]
Just pushed a few fixes -- let me try to improve test coverage too!
